### DISCUSSION TITLE
Parent not correctly identified as a class

### DISCUSSION THREAD

**Discussion by JordanLongstaff** [LemonAppDev/konsist#1616] [Report a Problem] [answered]
### Tell us what version of Konsist you use.

0.17.3

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

Here's a class I have declared:

```
abstract class SimpleEventPacket(reader: PacketReader) : Packet.Server(reader)
```

The only parent that Konsist finds has its source declaration cast to `KoExternalDeclarationCore` instead of `KoClassDeclarationCore`, so its fully qualified name is `Server` instead of the full path ending in `Packet.Server`. I'm guessing it's because `Packet.Server` is an inner class.

There seems to be a bug in the `sourceDeclaration` implementation:
```
val innerName = if (name.contains(".")) name.substringBeforeLast(".") else name
val outerName = if (name.contains(".")) name.substringAfterLast(".") else name
```

Shouldn't it be the other way around?

### Actual behaviour

The parent's `sourceDeclaration` property is cast to `KoExternalDeclarationCore`.

### Expected behaviour

The `sourceDeclaration` should be a `KoClassDeclarationCore`.

**Comment by JordanLongstaff** [comment]
Tests are failing in [this file](https://github.com/JordanLongstaff/ArtemisAgent/blob/renovate/konsist/IAN/packets/konsist/src/test/kotlin/com/walkertribe/ian/protocol/PacketKonsistTest.kt) in my repo. You can clone it if you want to debug.

**Reply by JordanLongstaff** [reply]
Except, now I've put in a workaround for the fix. My tests were failing because I was using `classes.withParentOf(Packet.Server::class)` and `packetClass.hasParentOf<Packet.Server>()`.

**Comment by nataliapeterwas** [comment]
Thank you for reporting the problem. We will take a look! We especially appreciate you sharing the project; it will be incredibly helpful for us.

**Comment by nataliapeterwas** [comment, marked as answer]
Will be fixed in next release: [PR](https://github.com/LemonAppDev/konsist/pull/1621)
